geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/kafka/commands/KeywordCommand.scala

```diff
@@ -0,0 +1,105 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.tools.kafka.commands
+
+import com.beust.jcommander._
+import com.beust.jcommander.converters.IParameterSplitter
+import org.locationtech.geomesa.kafka.{KafkaDataStore, KafkaDataStoreSchemaManager}
+import org.locationtech.geomesa.tools.kafka.commands.KeywordCommand.KeywordParameters
+import org.locationtech.geomesa.tools.common.OptionalFeatureTypeNameParam
+import org.locationtech.geomesa.tools.kafka.ProducerKDSConnectionParams
+import scala.collection.JavaConversions._
+import scala.io.StdIn
+import scala.util.{Try, Failure}
+
+class KeywordCommand(parent: JCommander) extends CommandWithKDS(parent) {
+
+  import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType._
+  override val command: String = "keyword"
+  override val params = new KeywordParameters()
+
+  override def execute(): Unit = {
+
+    if (ds == null) {
+      throw new ParameterException("Could not load a data store with the provided parameters")
+    }
+
+    val sft = ds.asInstanceOf[KafkaDataStore].getKafkaSchema(params.featureName)
+    var keywordsModified = false
+
+    if (params.keywordsToAdd != null) {
+      sft.addKeywords(params.keywordsToAdd.mkString(KEYWORDS_DELIMITER))
+      keywordsModified = true
+    }
+
+    if (params.keywordsToRemove != null) {
+      sft.removeKeywords(params.keywordsToRemove.mkString(KEYWORDS_DELIMITER))
+      keywordsModified = true
+    }
+
+    if (params.removeAll) {
+      val confirm = StdIn.readLine("Remove all keywords? (y/n): ").toLowerCase
+      if (confirm.equals("y") || confirm.equals("yes")) {
+        sft.removeAllKeywords()
+        keywordsModified = true
+      } else {
+        println("Aborting operation")
+        ds.dispose()
+        return
+      }
+    }
+
+    // Attempt to replace the old schema with the updated one
+    if (keywordsModified) {
+      Try {
+        ds.removeSchema(sft.getTypeName)
+      } match {
+        case Failure(ex) => println(ex.toString)
```

Comment:
org.I0Itec.zkclient.exception.ZkNodeExistsException: org.apache.zookeeper.KeeperException$NodeExistsException: KeeperErrorCode = NodeExists for /admin/delete_topics/geomesa-ds-kafka-example-json
	at org.I0Itec.zkclient.exception.ZkException.create(ZkException.java:55)
	at org.I0Itec.zkclient.ZkClient.retryUntilConnected(ZkClient.java:685)
	at org.I0Itec.zkclient.ZkClient.create(ZkClient.java:304)
	at org.I0Itec.zkclient.ZkClient.createPersistent(ZkClient.java:243)
	at kafka.utils.ZkUtils$.createPersistentPath(ZkUtils.scala:306)
	at kafka.admin.AdminUtils$.deleteTopic(AdminUtils.scala:159)
	at org.locationtech.geomesa.kafka08.ZkUtils08.deleteTopic(ZkUtils08.scala:23)
	at org.locationtech.geomesa.kafka.KafkaDataStoreSchemaManager$$anonfun$removeSchema$1.apply(KafkaDataStoreSchemaManager.scala:135)
	at org.locationtech.geomesa.kafka.KafkaDataStoreSchemaManager$$anonfun$removeSchema$1.apply(KafkaDataStoreSchemaManager.scala:133)
	at org.locationtech.geomesa.kafka.KafkaDataStoreSchemaManager$class.removeSchema(KafkaDataStoreSchemaManager.scala:133)
	at org.locationtech.geomesa.kafka.KafkaDataStore.removeSchema(KafkaDataStore.scala:28)
	at org.locationtech.geomesa.tools.kafka.commands.KeywordCommand.execute(KeywordCommand.scala:60)
	at org.locationtech.geomesa.tools.accumulo.commands.KeywordCommandTest$$anonfun$1$$anonfun$apply$3.apply(KeywordCommandTest.scala:84)
	at org.locationtech.geomesa.tools.accumulo.commands.KeywordCommandTest$$anonfun$1$$anonfun$apply$3.apply(KeywordCommandTest.scala:75)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:283)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:173)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:128)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:203)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:155)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
Caused by: org.apache.zookeeper.KeeperException$NodeExistsException: KeeperErrorCode = NodeExists for /admin/delete_topics/geomesa-ds-kafka-example-json
	at org.apache.zookeeper.KeeperException.create(KeeperException.java:119)
	at org.apache.zookeeper.KeeperException.create(KeeperException.java:51)
	at org.apache.zookeeper.ZooKeeper.create(ZooKeeper.java:783)
	at org.I0Itec.zkclient.ZkConnection.create(ZkConnection.java:87)
	at org.I0Itec.zkclient.ZkClient$1.call(ZkClient.java:308)
	at org.I0Itec.zkclient.ZkClient$1.call(ZkClient.java:304)
	at org.I0Itec.zkclient.ZkClient.retryUntilConnected(ZkClient.java:675)
	at org.I0Itec.zkclient.ZkClient.create(ZkClient.java:304)
	at org.I0Itec.zkclient.ZkClient.createPersistent(ZkClient.java:243)
	at kafka.utils.ZkUtils$.createPersistentPath(ZkUtils.scala:306)
	at kafka.admin.AdminUtils$.deleteTopic(AdminUtils.scala:159)
	at org.locationtech.geomesa.kafka08.ZkUtils08.deleteTopic(ZkUtils08.scala:23)
	at org.locationtech.geomesa.kafka.KafkaDataStoreSchemaManager$$anonfun$removeSchema$1.apply(KafkaDataStoreSchemaManager.scala:135)
	at org.locationtech.geomesa.kafka.KafkaDataStoreSchemaManager$$anonfun$removeSchema$1.apply(KafkaDataStoreSchemaManager.scala:133)
	at org.locationtech.geomesa.kafka.KafkaDataStoreSchemaManager$class.removeSchema(KafkaDataStoreSchemaManager.scala:133)
	at org.locationtech.geomesa.kafka.KafkaDataStore.removeSchema(KafkaDataStore.scala:28)
	at org.locationtech.geomesa.tools.kafka.commands.KeywordCommand.execute(KeywordCommand.scala:60)
	at org.locationtech.geomesa.tools.accumulo.commands.KeywordCommandTest$$anonfun$1$$anonfun$apply$3.apply(KeywordCommandTest.scala:84)
	at org.locationtech.geomesa.tools.accumulo.commands.KeywordCommandTest$$anonfun$1$$anonfun$apply$3.apply(KeywordCommandTest.scala:75)